dags/hivemind_etl_helpers/src/db/website/crawlee_client.py

```
@@ -0,0 +1,116 @@
+            if url and isinstance(url, str) and url.startswith(("http://", "https://")):
+                valid_links.append(url)
+            else:
+                raise ValueError(f"Invalid URL: {url}")
+
+        try:
+            await self.crawler.add_requests(requests=valid_links)
+            await asyncio.wait_for(self.crawler.run(), timeout=3600)  # 1 hour timeout
+            crawled_data = await self.crawler.get_data()
+            return crawled_data.items
+        except asyncio.TimeoutError:
+            raise TimeoutError("Crawl operation timed out")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)